pkg/module/validation.go

```diff
@@ -3,11 +3,348 @@ package module
 import (
 	"errors"
 	"fmt"
+	"io"
+	"os"
+	"path/filepath"
 	"strings"
+	"time"
 
+	"github.com/kyma-project/cli/internal/kube"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
 	amv "k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
 )
 
+const defaultCRName = "default.yaml"
+
+func ValidateDefaultCR(modulePath, envtestBinariesPath string, log *zap.SugaredLogger) (skipped bool, err error) {
+
+	exists, crData, err := readDefaultCR(modulePath)
+	if err != nil {
+		return false, err
+	}
+	if !exists {
+		//If there's no "default.yaml" file, skip the validation.
+		return true, nil
+	}
+
+	crMap, err := parseYamlToMap(crData)
+	if err != nil {
+		return false, fmt.Errorf("Error during parsing default CR: %w", err)
+	}
+
+	group, kind, err := readGroupKind(crMap)
+	if err != nil {
+		return false, err
+	}
+
+	searchDirPath := filepath.Join(modulePath, "charts")
+	crdFound, crdFilePath, err := findCRDFileFor(group, kind, searchDirPath)
+	if err != nil {
+		return false, fmt.Errorf("Error finding CRD file in the %q directory: %w", searchDirPath, err)
+	}
+	if !crdFound {
+		return false, fmt.Errorf("Can't find the CRD for CR instance defined in %q (group: %q, kind %q)", defaultCRName, group, kind)
+	}
+
+	err = ensureDefaultNamespace(crMap)
+	if err != nil {
+		return false, err
+	}
+
+	properCR, err := renderYamlFromMap(crMap)
+	if err != nil {
+		return false, err
+	}
+
+	envTest, restCfg, err := startValidationEnv(crdFilePath, envtestBinariesPath)
+	if err != nil {
+		return false, err
+	}
+
+	defer func() {
+		stopErr := envTest.Stop()
+		if stopErr != nil {
+			//TODO: This doesn't seem to print anything...
+			log.Error(fmt.Errorf("Error during stopping envTest: %w", stopErr))
+			//THIS does: fmt.Println(fmt.Errorf("Error during stopping envTest: %w", stopErr))
+		}
+	}()
+
+	kc, err := kube.NewFromRestConfigWithTimeout(restCfg, 30*time.Second)
+
+	err = kc.Apply(properCR)
+	if err != nil {
+		return false, fmt.Errorf("Error during applying the default CR: %w", err)
+	}
+	return false, nil
+}
+
+// readDefaultCR reads the default CR file's contents. The returned bool is true if the data is successfully read, it's false otherwise.
+func readDefaultCR(modulePath string) (bool, []byte, error) {
+	//TODO: Do we need to override the name or it's always "default.yaml"?
+	crPath := filepath.Join(modulePath, defaultCRName)
+
+	fileInfo, err := os.Stat(crPath)
+	if errors.Is(err, os.ErrNotExist) {
+		return false, nil, nil
+	}
+	if err != nil {
+		return false, nil, fmt.Errorf("Error accessing the default CR file `%q`: %w", crPath, err)
+	}
```

Comment:
You only need to check the kind of error if there is an error. This way is more readable as well:
```suggestion
	if err != nil {
		if errors.Is(err, os.ErrNotExist) {
			return false, nil, nil
		}
		return false, nil, fmt.Errorf("Error accessing the default CR file `%q`: %w", crPath, err)
	}
```